File: go/noreader.go
// noreader
package main

import (
	"flag"
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	//input := flag.String("input", "/home/gfeng/Desktop/History/Original", "input file or folder")
	//output := flag.String("output", "/home/gfeng/Desktop/History/Readable1", "save converted file to")
	input := flag.String("input", "", "input file or folder, ~/Desktop/History/Original")
	output := flag.String("output", "", "save converted file to, ~/Desktop/History/Readable")

	flag.Parse()

	fmt.Println("input = " + *input)
	fmt.Println("output = " + *output)

	fi, err := os.Stat(*input)
	if err != nil {
		fmt.Println(err)
		return
	}

	if _, err := os.Stat(*output); os.IsNotExist(err) {
		os.Mkdir(*output, os.ModePerm)
	}

	switch mode := fi.Mode(); {
	case mode.IsDir():
		fmt.Println("directory: " + *input)
		files, err := ioutil.ReadDir(*input)
		check(err)
		for _, file := range files {
			filename := filepath.Join(*input, file.Name())
			fmt.Println("file: " + filename)
			convert(filename, *output)
		}

	case mode.IsRegular():
		fmt.Println("file: " + *input)
		convert(*input, *output)
	}

	fmt.Println("Done!")
}

func convert(input string, output string) bool {
	data, err := ioutil.ReadFile(input)
	check(err)

	var outstr = ""

	for i := 0; i < len(data); i++ {
		var b = int(data[i])
		if data[i] < 0 {
			b = 255 + int(data[i])
		}

		if b > 31 && b < 127 || b == 10 || b == 13 {
			outstr += string(data[i])
		}
	}
	//fmt.Println(string(outstr))

	_, filename := filepath.Split(input)

	ioutil.WriteFile(filepath.Join(output, filename), []byte(outstr), os.ModePerm)

	return true
}

File: java/src/main/java/NOReader.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.PrintWriter;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.stream.Stream;

public class NOReader {

	public static void main(String[] args) throws Exception {
		String input = "/home/gfeng/Desktop/History/Original";
		String output = "/home/gfeng/Desktop/History/Readable";
		Stream<Path> stream = Files.list(Paths.get(input));

		stream.forEach(t -> {
			File in = t.toFile();
			File out = new File(output, in.getName());

			try {
				convert(in, out);
				System.out.println("Done: " + out);
			} catch (Exception e) {
				e.printStackTrace();
			}
		});
	}

	private static void convert(File in, File out) throws Exception {
		try (FileInputStream reader = new FileInputStream(in); PrintWriter writer = new PrintWriter(out)) {
			byte[] buf = new byte[1024];

			int len = 0;

			while ((len = reader.read(buf)) > -1) {
				for (int i = 0; i < len; i++) {
					byte b = buf[i] > -1 ? buf[i] : (byte) (buf[i] & 0xFF);

					if (b > 31 && b < 127 || b == 10 || b == 13) {
						writer.write(String.valueOf((char) b));
						System.out.print((char) b);
					}

				}
			}
		}
	}
}